src/ray/common/status_or.h

```diff
@@ -0,0 +1,261 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <stdexcept>
+#include <string_view>
+#include <type_traits>
+#include <utility>
+
+#include "absl/base/attributes.h"
+#include "ray/common/status.h"
+
+namespace ray {
+
+template <typename T>
+class StatusOr {
```

Comment:
For ergonomics, we may also want common StatusOr macros:

ASSIGN_OR_RETURN
RETURN_IF_ERROR

Reply:
Exactly, I will make it in the next PR.
It's definitely on my list :)